### PR TITLE
[$50] - Next Deadline msg on Challenge Details page for past challenges #445

### PR DESCRIPTION
**Pull request by imadx**

### CODE REVIEW

File: src/shared/components/Terms/index.jsx [hunk 1]
```diff
@@ -160,7 +160,7 @@ export default class Terms extends React.Component {
           {
             !isLoadingTerms && (
               <div styleName="modal-content">
-                <div styleName="title">{terms.length > 1 ? 'Terms & Conditions of Use' : terms[0].title}</div>
+                <div styleName="title">{terms.length > 1 ? 'Challenge Prerequisites' : terms[0].title}</div>
```
Comment (birdofpreyru) [inline src/shared/components/Terms/index.jsx]: This change seems to be related to another issue.

File: src/shared/components/Terms/styles.scss [hunk 2]
```diff
@@ -44,6 +44,7 @@
   font-size: 28px;
   margin-bottom: 1.5 * $base-unit;
   text-align: center;
+  color: $tc-gray-80;
```
Comment (birdofpreyru) [inline src/shared/components/Terms/styles.scss]: This as well related to another issue, please remove from this PR.

File: src/shared/components/challenge-detail/Header/index.jsx [hunk 3]
```diff
@@ -187,12 +189,20 @@ export default function ChallengeHeader(props) {
   let nextDeadlineMsg;
   switch ((status || '').toLowerCase()) {
     case 'active':
-      nextDeadlineMsg = (
-        <div styleName="next-deadline">
-          Next Deadline: <span styleName="deadline-highlighted">
-            {nextDeadline || '-'}</span>
-        </div>
-      );
+      if (timeLeft === 0) {
```
Comment (birdofpreyru) [inline src/shared/components/challenge-detail/Header/index.jsx]: This does not look correct to me. To check whether a challenge is stalled, we should look at the status field in the challenge object.

File: src/shared/components/Terms/index.jsx (hunk 1 above)
Comment (imadx) [reply src/shared/components/Terms/index.jsx]: My bad... I'll remove this.

File: src/shared/components/Terms/styles.scss (hunk 2 above)
Comment (imadx) [reply src/shared/components/Terms/styles.scss]: Will remove this as well.

File: src/shared/components/challenge-detail/Header/index.jsx (hunk 3 above)
Comment (imadx) [reply src/shared/components/challenge-detail/Header/index.jsx]: do we have a `stalled` status along with `active` and `completed` states?
where can I find a challenge in `stalled` competition to test?

Comment (birdofpreyru) [reply src/shared/components/challenge-detail/Header/index.jsx]: @imadx I guess so, but I might be wrong. There are many stalled F2F in the dev challenge listing, and you can find out in the listing code, what is the actual conidition to distinguish stalled challenges.

Comment (imadx) [reply src/shared/components/challenge-detail/Header/index.jsx]: @birdofpreyru The stalled competitions also shows their status to be `active`.
It doesn't seem to be the number of registrants either. Do you know anyone who knows what distinguishes stalled challenges?

Comment (birdofpreyru) [reply src/shared/components/challenge-detail/Header/index.jsx]: Ok, I checked it myself, it is all here: https://github.com/topcoder-platform/community-app/blob/develop/src/shared/components/challenge-listing/ChallengeCard/Status/index.jsx
A challenge is stalled if it has no open phases beside registration phase.

Comment (imadx) [reply src/shared/components/challenge-detail/Header/index.jsx]: Awesome! I'll fix this...

Comment (imadx) [reply src/shared/components/challenge-detail/Header/index.jsx]: @birdofpreyru made the changes and checked with the challenge listing.
I hope the behavior is fixed now.

**Comment by birdofpreyru** [comment]
I've checked against dev backend, there are many challenges that are late, but the message just shows 0 till the end of the phase, e.g.:

**Comment by ishan-madhusanka** [comment]
Was that not what was needed? What was the required behavior?

**Comment by birdofpreyru** [comment]
Have you read my *Expected Fix* post in the issue ticket?  For challenges being in a phase that is late to close we write `Late by ...` similar to how it is done in the challenge listing.

**Comment by imadx** [comment]
@birdofpreyru Added the PR for this missing piece at https://github.com/topcoder-platform/community-app/pull/735

**Pull request closed without merging**